tests/python_client/testcases/test_search.py

```diff
@@ -3062,6 +3062,34 @@ def test_search_output_vector_field_and_scalar_field(self):
         # 3. check the result
         assert default_float_field_name, default_string_field_name in res[0][0].entity._row_data
         assert default_search_field in res[0][0].entity._row_data
+        assert len(res[0][0].entity._row_data) == 3
+
+    @pytest.mark.tags(CaseLabel.L2)
+    def test_search_output_vector_field_and_pk_field(self):
+        """
+        target: test search with output vector field and pk field
+        method: 1. initialize a collection
+                2. search with output field vector
+                3. check no field missing
+        expected: search success
+        """
+        # 1. initialize a collection
+        collection_w = self.init_collection_general(prefix, True)[0]
+
+        # 2. search with output field vector
+        res = collection_w.search(vectors[:1], default_search_field,
+                                  default_search_params, default_limit, default_search_exp,
+                                  output_fields=[default_int64_field_name,
+                                                 default_string_field_name,
+                                                 default_search_field],
+                                  check_task=CheckTasks.check_search_results,
+                                  check_items={"nq": 1,
+                                               "limit": default_limit})[0]
+
+        # 3. check the result
+        assert default_int64_field_name, default_string_field_name in res[0][0].entity._row_data
+        assert default_search_field in res[0][0].entity._row_data
+        assert len(res[0][0].entity._row_data) == 3
```

Comment:
Only length check is not enough either.

Try out_put = output_fields=[default_int64_field_name,
                                                 default_string_field_name,
                                                 default_search_field],

And out_put = output_fields=[default_string_field_name,
                                                 default_search_field],

And compare the results, you could know why.

And suggest to write an output_fields results check function to replace all the current check.

Reply:
I will add the output_fields check function in another pr. Cause i found it hard to get the entity data. #24014